samples/server/petstore/rust-server/output/openapi-v3/src/server/context.rs

```diff
@@ -83,6 +83,16 @@ impl<T, A, B, C, D> hyper::server::Service for AddContext<T, A>
     fn call(&self, req: Self::Request) -> Self::Future {
         let context = A::default().push(XSpanIdString::get_or_generate(&req));
 
+        {
+            use hyper::header::{Authorization as HyperAuth, Basic, Bearer};
+            use std::ops::Deref;
+            if let Some(bearer) = req.headers().get::<HyperAuth<Bearer>>().cloned() {
+                let auth_data = AuthData::Bearer(bearer.deref().clone());
+                let context = context.push(Some(auth_data));
+                let context = context.push(None::<Authorization>);
+                return self.inner.call((req, context));
+            }
```

Comment:
Just want to check that this is intentional. Is it OK to do nothing if we don't have a bearer header? Or am I misreading how this works?

Reply:
It's definitely fine here - because this code is API operation agnostic, and the authentication requirements may be different on a per operation basis.

An absent bearer will either be handled by the OpenAPI scope code (rejecting an absent bearer) or by the scope checking code (which will reject the request as it doesn't have sufficient scopes on it).

The former code isn't provided by the auto-generated code, the later is provided by https://github.com/OpenAPITools/openapi-generator/pull/3495/files/b1df259d3824c7de0b27feac684d5f5bd4577d6e#diff-1214066435f929c4d1bc643c36364470R143 where we check whether the Authorization scopes match.